lightning_client.go

```diff
@@ -1087,6 +1087,11 @@ type Payment struct {
 	// if the payment is settled.
 	Preimage *lntypes.Preimage
 
+	// PaymentRequest is the payment request for this payment. This value
+	// will not be set for keysend payments and for payments that manually
+	// specify their destination and amount.
+	PaymentRequest string
```

Comment:
Do we need this somewhere? I always had doubts about exposing this even.

Reply:
I need this in Faraday to detect circular payments. Main logic uses htlc attempts, but legacy payments don't have htlcs, so we're going to fall back to payment request. 

It's not ideal, since some legacy payments won't have pay_req either, but I'd rather cover as much of the edge case as I can.

Reply:
Ok, yes if it is only for legacy it seems no problem